client/src/components/RepoOverview.tsx

```diff
@@ -1,16 +1,66 @@
-import React, { useEffect } from 'react';
-import { RepoDTO } from '../api/repo.api';
+import React, { useEffect, useState } from 'react';
+import { RepoExtDTO, RepositoryService } from '../api/repo.api';
+import { RepositoryDescUpdateDTO } from '../api/repo.api';
+import { ToastType, useToastStore } from '../util/toastStore';
+import { Alert } from 'react-bootstrap';
+import { AxiosError } from 'axios';
+
+export const RepoOverview: React.FC<{ isActive: boolean; repo: RepoExtDTO, repoStateChanger: any }> = (props) => {
+    const [isEditing, setIsEditing] = useState(false);
+    const [newDesc, setNewDesc] = useState(props.repo.desc);
+    const addToast = useToastStore((state) => state.addToast);
+    const [error, setError] = useState('');
 
-export const RepoOverview: React.FC<{ isActive: boolean, repo: RepoDTO }> = (props) => {
     useEffect(() => {
         if (props.isActive) {
             // Tab selected, do stuff here...
         }
     }, [props.isActive]);
 
+    const updateChanges = () => {
+        setIsEditing(false);
+
+        let dto: RepositoryDescUpdateDTO = {
+            desc: newDesc,
+        };
+        
+        RepositoryService.UpdateRepoDescById(props.repo.id, dto).then((res) => {
+            props.repoStateChanger({
+                ...props.repo,
+                ...res.data,
+            });
+            addToast(`Updated repository ${props.repo.name}`, ToastType.success);
+            setError('');
```

Comment:
Since this function exclusively updates the description of the repository, make the toast message more descriptive as well ("Updated the description of {reponame}" or something like that).